### PR TITLE
feat(lidar_centerpoint): single inference lidar centerpoint node

### PR DESCRIPTION
**Pull request by knzo25**
## Description
This PR implements a separate lidar center point node and its launcher to perform single inferences.

It loads a single pointcloud from a launcher parameters and outputs a `ply` file containing the detections' results as meshes. The visualization of the results can be visualized as follows:

```
import open3d as o3d

mesh = o3d.io.read_triangle_mesh("test.ply")
pcd = o3d.io.read_point_cloud("test.pcd")

mesh_frame = o3d.geometry.TriangleMesh.create_coordinate_frame(
    size=1.0, origin=[0, 0, 0])

detection_lines = o3d.geometry.LineSet.create_from_triangle_mesh(mesh)
detection_lines.paint_uniform_color([1.0, 0.0, 1.0])

o3d.visualization.draw_geometries(
    [mesh_frame, pcd, detection_lines])

```

## Related links

[TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-19941)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by knzo25** [comment]
This PR also contains the code from #1925, since we needed to avoid the tf dependency. As such, it would be better to review and merge that PR first.

**Comment by yukke42** [comment]
@knzo25 Cloud you add the python script to load the results and describe the usage in README?

**Comment by knzo25** [comment]
@yukke42 
Done !

**Comment by yukke42** [comment]
@knzo25 Cloud you fix to pass to build CI?

**Comment by yukke42** [comment]
@knzo25 Cloud you fix to pass build CI?
https://github.com/autowarefoundation/autoware.universe/actions/runs/3520503806/jobs/5901616068

File: perception/lidar_centerpoint/README.md [hunk 1]
```diff
@@ -52,6 +52,22 @@ We trained the models using <https://github.com/open-mmlab/mmdetection3d>.
...
+```bash
+ros2 launch lidar_centerpoint single_inference_lidar_centerpoint.launch.xml pcd_path:=test_pointcloud.pcd
+```
+
+`lidar_centerpoint` generated a `ply` file with the same name which contains the detections as meshes.
+These detections can be visualized by most 3D tools, but we also provide a small script to visualize the detections alonsgside the input pointcloud.
+
+```bash
+python3 scripts/visualize_resuls.py -pcd_path test_pointcloud.pcd -detections_path test_pointcloud.ply
+```
```
Comment (yukke42) [inline perception/lidar_centerpoint/README.md:69]: It's better to add visualization script to the launch file for the simple command.

File: perception/lidar_centerpoint/scripts/visualize_resuls.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/ usr / bin / env python3
```
Comment (yukke42) [inline perception/lidar_centerpoint/scripts/visualize_resuls.py:1]:
```suggestion
#!/usr/bin/env python3
```

File: perception/lidar_centerpoint/scripts/visualize_resuls.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# Copyright 2020 Tier IV, Inc.
+#
+# Licensed under the Apache License, Version 2.0(the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:  // www.apache.org/licenses/LICENSE-2.0
```
Comment (yukke42) [inline perception/lidar_centerpoint/scripts/visualize_resuls.py:9]:
```suggestion
#   http://www.apache.org/licenses/LICENSE-2.0
```

File: perception/lidar_centerpoint/README.md (hunk 1 above)
Comment (knzo25) [reply perception/lidar_centerpoint/README.md:69]: @yukke42 
How about now?
The visualization script is executed together with the inference node

Comment (yukke42) [reply perception/lidar_centerpoint/README.md:69]: looks good! please update docs too.

File: perception/lidar_centerpoint/src/single_inference_node.cpp [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+#include <lidar_centerpoint/preprocess/pointcloud_densification.hpp>
+#include <lidar_centerpoint/ros_utils.hpp>
+#include <lidar_centerpoint/utils.hpp>
+#include <pcl_ros/transforms.hpp>
+
+#include <sensor_msgs/msg/point_cloud2.hpp>
+
+#include <pcl/io/file_io.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <tf2_eigen/tf2_eigen.h>
```
Comment (yukke42) [inline perception/lidar_centerpoint/src/single_inference_node.cpp:27]:
```suggestion
#ifdef ROS_DISTRO_GALACTIC
#include <tf2_eigen/tf2_eigen.h>
#else
#include <tf2_eigen/tf2_eigen.hpp>
#endif
```

File: perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
```
Comment (yukke42) [inline perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py:2]:
```suggestion
```
The copyright might be in the first line to pass CI test.

File: perception/lidar_centerpoint/src/single_inference_node.cpp [hunk 6]
```diff
@@ -0,0 +1,244 @@
+// Copyright 2021 TIER IV, Inc.
```
Comment (yukke42) [inline perception/lidar_centerpoint/src/single_inference_node.cpp:1]:
```suggestion
// Copyright 2022 TIER IV, Inc.
```

File: perception/lidar_centerpoint/README.md (hunk 1 above)
Comment (yukke42) [reply perception/lidar_centerpoint/README.md]: [794a49e](https://github.com/autowarefoundation/autoware.universe/pull/1926/commits/794a49e3c2432c526b8f3e517882fb77024d1ab7)

**Comment by knzo25** [comment]
@yukke42 
Added the documentation and fixed the license header again (copied the one from other file and it passed this time)

File: perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Tier IV, Inc. All rights reserved.
```
Comment (yukke42) [inline perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py:1]:
```suggestion
# Copyright 2022 TIER IV, Inc. All rights reserved.
```

**Comment by yukke42** [comment]
@knzo25 Cloud you share a sample pcd file to review?

**Comment by knzo25** [comment]
> @knzo25 Cloud you share a sample pcd file to review?

Doesn't the one I posted in the ticket at the beginning work?

**Comment by yukke42** [comment]
> > @knzo25 Cloud you share a sample pcd file to review?
> 
> Doesn't the one I posted in the ticket at the beginning work?

I see. It helps me to describe it in the description since I cannot remember all contents of the tickets. 
Cloud you add the ticket link as "TIER IV INTERNAL LINK" for the integration team?

**Comment by knzo25** [comment]
> > > @knzo25 Cloud you share a sample pcd file to review?
> > 
> > 
> > Doesn't the one I posted in the ticket at the beginning work?
> 
> I see. It helps me to write it in the description since I cannot remember all contents of the tickets. Cloud you add the ticket link as "TIER IV INTERNAL LINK" for the integration team?

Done !

File: perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 TIER IV, Inc. All rights reserved.
```
Comment (yukke42) [inline perception/lidar_centerpoint/scripts/lidar_centerpoint_visualizer.py:1]:
```suggestion
#!/usr/bin/env python
# Copyright 2022 TIER IV, Inc. All rights reserved.
```

**Review by yukke42 — APPROVED** [review]
LGTM

**Comment by knzo25** [comment]
:tada:

**Pull request merged**